core/src/Cuda/Kokkos_Cuda_Team.hpp

```diff
@@ -186,22 +186,23 @@ class CudaTeamMember {
   template <typename ReducerType>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<is_reducer<ReducerType>::value>
   team_reduce(ReducerType const& reducer) const noexcept {
-    team_reduce(reducer, reducer.reference());
+    using value_type = typename ReducerType::value_type;
+    using wrapped_reducer_type =
+        typename Impl::FunctorAnalysis<Impl::FunctorPatternInterface::REDUCE,
+                                       TeamPolicy<Cuda>, ReducerType,
+                                       value_type>::Reducer;
+
+    team_reduce(wrapped_reducer_type(reducer), reducer.reference());
   }
 
-  template <typename ReducerType>
+  template <typename ReducerType, typename ValueType>
   KOKKOS_INLINE_FUNCTION std::enable_if_t<is_reducer<ReducerType>::value>
-  team_reduce(ReducerType const& reducer,
-              typename ReducerType::value_type& value) const noexcept {
+  team_reduce(ReducerType const& reducer, ValueType& value) const noexcept {
```

Comment:
What if `typename ReducerType::value_type` and `ValueType` don't match?

Reply:
Reverted to use `ReducerType::value_type`.